sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/EnvironmentResource.java

```diff
@@ -5,22 +5,28 @@
 
 package io.opentelemetry.sdk.autoconfigure;
 
+import io.opentelemetry.api.common.AttributeKey;
 import io.opentelemetry.api.common.Attributes;
 import io.opentelemetry.api.common.AttributesBuilder;
 import io.opentelemetry.sdk.resources.Resource;
 
 /**
  * Factory for a {@link Resource} which parses the standard "otel.resource.attributes" system
- * property or OTEL_RESOURCE_ATTRIBUTES environment variable.
+ * property or OTEL_RESOURCE_ATTRIBUTES environment variable. Will also use
+ * OTEL_SERVICE_NAME/otel.service.name to specifically set the service name.
  */
 public final class EnvironmentResource {
 
   // Visible for testing
   static final String ATTRIBUTE_PROPERTY = "otel.resource.attributes";
+  static final String SERVICE_NAME_PROPERTY = "otel.service.name";
+  // when semconv is stable, this should use ResourceAttributes.SERVICE_NAME
```

Comment:
Since it's private usage we can use semconv now right?

Reply:
we don't currently depend on semconv, except for tests in this module. Do we want to?

Reply:
Yeah since we depend on metrics, same story I think

Reply:
fair enough. done!